src/player.rs

```diff
@@ -0,0 +1,297 @@
+            // Now its the same as for collision with bullet,
+            // but in the futer calculating damage amount might be different
+            damage_writer.send(TakeDamageEvent {
+                id: player,
+                amount: 1,
+            });
+        }
+    }
+}
+
+fn player_dies(
+    mut commands: Commands,
+    mut death_reader: EventReader<DeathEvent>,
+    mut players: Query<(Entity), (With<PlayerMarker>)>,
+) {
+    for death in death_reader.iter() {
+        if let Ok(plyer) = players.get(death.id) {
+            info!("Zed's Dead");
+        }
+    }
+}
+
+pub fn control_bullets(
+    mut commands: Commands,
+    mut player: Query<&Transform, With<PlayerMarker>>,
+    mut timer: ResMut<Timer>,
+    keyboard_input: Res<Input<KeyCode>>,
+    time: Res<Time>,
+) {
+    timer.tick(time.delta());
+    if let Ok(mut player_tf) = player.get_single_mut() {
+        let shoot_direction = get_direction_from_keyboard(
+            &keyboard_input,
+            KeyCode::Up,
+            KeyCode::Down,
+            KeyCode::Left,
+            KeyCode::Right,
+        );
+
+        if shoot_direction != Vec2::new(0., 0.) {
+            let mut spawn_bullets = || {
+                commands
+                    .spawn_bundle(BulletBundle::new(
+                        hitbox::Hitbox::new_rectangle(Vec2::new(7., 25.)),
+                        shoot_direction * BULLET_START_SPEED,
+                    ))
+                    .insert_bundle(SpriteBundle {
+                        sprite: Sprite {
+                            color: Color::from([30., 130., 240.]),
+                            custom_size: Some(Vec2::new(7., 25.)),
+                            ..Default::default()
+                        },
+                        transform: Transform {
+                            translation: Vec3::new(
+                                player_tf.translation.x,
+                                player_tf.translation.y,
+                                2.0,
+                            ),
+                            ..Default::default()
+                        },
+                        ..Default::default()
+                    });
+            };
+
+            if timer.finished() {
+                spawn_bullets();
+                timer.reset();
+            }
+        }
+    }
+}
+
+fn bullet_movement(
+    mut commands: Commands,
+    win_size: Res<WinSize>,
+    mut query: Query<
+        (
+            Entity,
+            &DirectionVector,
+            &mut VelocityVector,
+            &Speed,
+            &Transform,
+        ),
+        (With<BulletMarker>),
+    >,
+) {
+    for (bullet_entity, direction, mut vel, &Speed(speed), bullet_tf) in query.iter_mut() {
+        let new_vel = direction.0 * speed;
+        vel.0 += new_vel;
+
+        // Delete entity if it's outside of the window.
+        let translation = &bullet_tf.translation;
+        if translation.y > win_size.height || translation.x > win_size.width {
```

Comment:
pomieszanie porządków: co mają wymiary okna do świata gry? jak sobie powiększę okno to moje pociski nie powinny móc lecieć dalej. No i w ogóle powinniście wprowadzić tłumaczenie pozycja w grze -> pozycja na ekranie, bo póki co macie tylko jedną mapę wielkości domyślnej wielkości okna, ale przecież zaraz tak nie będzie...